cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/BalancingAction.java

```diff
@@ -65,18 +66,46 @@ public BalancingAction(TopicPartition sourceTp,
   }
 
   /**
-   * Constructor for creating a balancing proposal with given topic partitions, source and destination broker id and logdir,
-   * and the topic partition of replica to swap with.
+   * Constructor for creating a balancing proposal with given topic partition, source and destination disk, and
+   * balancing action type.
    *
-   * @param sourceTp                Topic partition of the source replica.
-   * @param sourceBrokerId          Source broker id of the replica.
-   * @param sourceBrokerLogdir      Source broker logdir of the replica.
-   * @param destinationBrokerId     Destination broker id of the replica.
-   * @param destinationBrokerLogdir Destination broker logdir of the replica.
-   * @param actionType              Action type.
-   * @param destinationTp           Topic partition of the replica to swap with.
+   * @param tp              Topic partition of the replica.
+   * @param sourceDisk      Source disk of the replica.
+   * @param destinationDisk Destination disk of the replica.
+   * @param actionType      Action type.
+   */
+  public BalancingAction(TopicPartition tp,
+                         Disk sourceDisk,
+                         Disk destinationDisk,
+                         ActionType actionType) {
+    this(tp, sourceDisk.broker().id(), sourceDisk.logDir(), destinationDisk.broker().id(), destinationDisk.logDir(), actionType, tp);
+  }
+
+  /**
+   * Constructor for creating a balancing proposal with given topic partitions, source and destination disk, balancing
+   * action type and the topic partition of replica to swap with.
```

Comment:
If the intention is to allow only `swap` as indicated in the JavaDoc, why do we get an `actionType`? If not, can we indicate the other cases? e.g. when is it `null`?